cmd/publishing-bot/config/rules.go

```diff
@@ -68,15 +70,21 @@ type RepositoryRules struct {
 
 // LoadRules loads the repository rules either from the remote HTTP location or
 // a local file path.
-func LoadRules(ruleFile string) (*RepositoryRules, error) {
+func LoadRules(ruleFile, sourceBranch, repoDir string) (*RepositoryRules, error) {
 	var (
 		content []byte
 		err     error
 	)
+
 	if ruleUrl, err := url.ParseRequestURI(ruleFile); err == nil && len(ruleUrl.Host) > 0 {
+		// The sourceBranch will be ignored when reading from URL
 		content, err = readFromUrl(ruleUrl)
 	} else {
-		content, err = ioutil.ReadFile(ruleFile)
+		if len(sourceBranch) > 0 {
```

Comment:
can we get rid of this if clause by defaulting the sourceBranch to "master" outside of this func? Actually, I would split this function up. We model a sum type in the input by multiple parameters. That's ugly.